### ISSUE TITLE
Chaining configuration blocks via 'extends' throws exception

### ISSUE THREAD

**Issue by lahmann** [acelaya/ZF2-AcMailer#122]
When chaining multiple configuration blocks via `extends`...

```
return [
    'acmailer_options' => [
        'default' => [
            //unaltered default values from mail.global.php.dist
        ],
        'foo' => [
            'extends' => 'default',
            'message_options' => [
                'from' => 'me@here.com'
            ]
        ],
        'bar' => [
            'extends' => 'foo',
            'message_options' => [
                'to' => 'me@here.com',
                'subject' => 'Foobar subject'
            ]
        ]
    ],
];
```

...AcMailer will 

* work fine with `$sm->get('acmailer.mailservice.foo')`
* but breaks with `$sm->get('acmailer.mailservice.bar')` showing exception:

```
Zend\Stdlib\Exception\BadMethodCallException
The option "extends" does not have a callable "setExtends" ("setextends") setter method which must be defined
#0 /usr/local/vufind2/vendor/zendframework/zendframework/library/Zend/Stdlib/AbstractOptions.php(62): Zend\Stdlib\AbstractOptions->__set('extends', 'default')
#1 /usr/local/vufind2/vendor/zendframework/zendframework/library/Zend/Stdlib/AbstractOptions.php(32): Zend\Stdlib\AbstractOptions->setFromArray(Array)
#2 /usr/local/vufind2/vendor/acelaya/zf2-acmailer/src/Options/Factory/MailOptionsAbstractFactory.php(54): Zend\Stdlib\AbstractOptions->__construct(Array)
#3 [internal function]: AcMailer\Options\Factory\MailOptionsAbstractFactory->createServiceWithName(Object(Zend\ServiceManager\ServiceManager), 'acmailer.mailop...', 'acmailer.mailop...')
(...)
#29 {main}
```

**Comment by acelaya** [comment]
Thanks, I'll take a look at it.

--
Alejandro Celaya Alastrué

**Comment by lahmann** [comment]
solved with PR #123
